File: Programmers/Python/Code/실패율.py
# 풀이1
def solution(N, stages):
    answer = []
    ss = sorted(stages)
    total = len(ss)
    temp = []
    for i in range(1, N+1):
        if i in ss:
            value = ss.count(i)
            temp.append(value/total)
            total -= value
        else:
            temp.append(0)
    # print(temp)
    for i in range(N):
        pop_idx = temp.index(max(temp))
        answer.append(pop_idx+1)
        temp[pop_idx] = -1

    return answer

print(solution(5, [2, 1, 2, 6, 2, 4, 3, 3]))

# sorted(result, key=lambda x : result[x], reverse=True)

File: Programmers/JavaScript/Code/실패율.js
function solution(N, stages) {
    var result = {};
    var total = stages.length
    // stages(1~N)를 탐색하면서
    for (var stage = 1; stage < N+1; stage++) {
        // 현재 스테이지에 총원이 0이 아니면
        if (total != 0) {
            // 현재 스테이지에 사람이 몇명인지 확인
            let count = stages.filter(e => stage === e).length;
            // 실패율 구하기
            result[stage] = count / total
            // 총원 빼기
            total -= count
        }
        // 총원 0명에 아무도 못깼으면
        else {
            result[stage] = 0
        }
    }
    // console.log(result)

    // 
    let sortobj = [];
    for (let number in result) {
        sortobj.push([number, result[number]]);
    }
    sortobj.sort((a, b) => b[1] - a[1]);


    return sortobj.map(a => parseInt(a[0]));
}

var N = 5
var stages = [2,1,2,4,2,4,3,3]

console.log(solution(N, stages));
